R/addins.R

```diff
@@ -10,7 +10,7 @@ addin_lint <- function() {
     config_linters <- NULL
   } else {
     config <- read.dcf(config_file, all = TRUE)
-    config_linters <- gsub("\n", "", config[["linters"]])
+    config_linters <- gsub("(#.*?)?\n", "", config[["linters"]])
```

Comment:
Do we need to `gsub()` at all?
The new regex will, as you correctly pointed out, break if `#` is contained in strings or symbol names.

Reply:
No, I've done a couple of tests and I don't think the `gsub()` is needed, so I removed it in a new commit.

Until now I thought the whole reading/parsing of the config file in this function wasn't necessary, but I found one  'scenario' for which the parsing needs to be done here. Currently, if you have a `.lintr` file without a linters section, you get the "No configuration found. Using default linters." message. To be able to show the message, the config needs to be parsed here.